### PR TITLE
[SRE-117] hasura 1.2.1 upgrade

### PR DESCRIPTION
**Pull request by craigem**
* Hasura 1.2.1 upgrade
* Addition of `hasura-cli`
* Addition of `regex-tdfa`

### CODE REVIEW

**Comment by craigem** [comment]
Rebased on master.

**Comment by rhyslbw** [comment]
#186 makes this easier. Once merged you can rebase to get access to the migration script

**Comment by craigem** [comment]
Relies on [PR #186](https://github.com/input-output-hk/cardano-graphql/pull/186) being merged and this branch rebased.

**Comment by rhyslbw** [comment]
@craigem 186 has been merged

File: nix/cardano-graphql.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+, cardano-graphql-src
+, runtimeShell
+, nodejs
+, python
+}:
+
+mkYarnPackage {
+  pname = "cardano-graphql";
+  name = "cardano-graphql";
+  version = "0.4.0";
```
Comment (rhyslbw) [inline nix/cardano-graphql.nix:12]: I thought this was being referenced from the `package.json` ?

File: nix/cardano-graphql.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  yarnPreBuild = ''
+    mkdir -p $HOME/.node-gyp/${nodejs.version}
+    echo 9 > $HOME/.node-gyp/${nodejs.version}/installVersion
+    ln -sfv ${nodejs}/include $HOME/.node-gyp/${nodejs.version}
+  '';
+  pkgConfig = {
+    node-sass = {
+      buildInputs = [ python ];
+      postInstall = ''
+        yarn --offline run build
```
Comment (cleverca22) [inline nix/cardano-graphql.nix:48]: is `node-sass` actually used by hasura?

File: nix/cardano-graphql.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      '';
+    };
+  };
+
+  installPhase = ''
+    unpackPhase
+    cd $sourceRoot
```
Comment (cleverca22) [inline nix/cardano-graphql.nix:55]: why the `unpackPhase` here? it should already have been unpacked by previous phases

**Pull request merged**